ui-v2/src/hooks/global-concurrency-limits.test.tsx

```diff
@@ -112,4 +117,202 @@ describe("global concurrency limits hooks", () => {
 		await waitFor(() => expect(result.current.isSuccess).toBe(true));
 		expect(result.current.data).toEqual(mockDetails);
 	});
+
+	/**
+	 * Data Management:
+	 * - Asserts global concurrency limit calls delete API and refetches updated list
+	 */
+	it("useDeleteGlobalConcurrencyLimit() invalidates cache and fetches updated value", async () => {
+		const ID_TO_DELETE = "0";
+		const queryClient = new QueryClient();
+
+		// ------------ Mock API requests after queries are invalidated
+		const mockData = seedGlobalConcurrencyLimits().filter(
+			(limit) => limit.id !== ID_TO_DELETE,
+		);
+		mockFetchGlobalConcurrencyLimitsAPI(mockData);
+
+		// ------------ Initialize cache
+		queryClient.setQueryData(
+			buildListGlobalConcurrencyLimitsQuery(filter).queryKey,
+			seedGlobalConcurrencyLimits(),
+		);
+
+		// ------------ Initialize hooks to test
+		const { result: useListGlobalConcurrencyLimitsResult } = renderHook(
+			() => useListGlobalConcurrencyLimits(filter),
+			{ wrapper: createQueryWrapper({ queryClient }) },
+		);
+
+		const { result: useDeleteGlobalConcurrencyLimitResult } = renderHook(
+			useDeleteGlobalConcurrencyLimit,
+			{ wrapper: createQueryWrapper({ queryClient }) },
+		);
+
+		// ------------ Invoke mutation
+		act(() =>
+			useDeleteGlobalConcurrencyLimitResult.current.deleteGlobalConcurrencyLimit(
+				ID_TO_DELETE,
+			),
+		);
+
+		// ------------ Assert
+		await waitFor(() =>
+			expect(useDeleteGlobalConcurrencyLimitResult.current.isSuccess).toBe(
+				true,
+			),
+		);
+		expect(useListGlobalConcurrencyLimitsResult.current.data).toHaveLength(0);
+	});
+
+	/**
+	 * Data Management:
+	 * - Asserts create mutation API is called.
+	 * - Upon create mutation API being called, cache is invalidated and asserts cache invalidation APIS are called
+	 */
+	it("useCreateGlobalConcurrencyLimit() invalidates cache and fetches updated value", async () => {
+		const queryClient = new QueryClient();
+		const MOCK_NEW_LIMIT_ID = "1";
+		const MOCK_NEW_LIMIT = {
+			active: true,
+			active_slots: 0,
+			denied_slots: 0,
+			limit: 0,
+			name: "global concurrency limit 1",
+			slot_decay_per_second: 0,
+		};
+
+		// ------------ Mock API requests after queries are invalidated
+		const NEW_LIMIT_DATA = {
+			...MOCK_NEW_LIMIT,
+			id: MOCK_NEW_LIMIT_ID,
+			created: "2021-01-01T00:00:00Z",
+			updated: "2021-01-01T00:00:00Z",
+			active_slots: 0,
+			slot_decay_per_second: 0,
+		};
+
+		const mockData = [
+			...seedGlobalConcurrencyLimits(),
+			{
+				...MOCK_NEW_LIMIT,
+				...NEW_LIMIT_DATA,
+			},
```

Comment:
I think the `MOCK_NEW_LIMIT` spread is duplicated here? It looks like it's already included in `NEW_LIMIT_DATA`.

Reply:
Thanks for the catch. Updated in latest push